pkg/k8shandler/clusterrestart.go

```diff
@@ -543,9 +556,15 @@ func (r Restarter) restartCluster() error {
 	}
 
 	if r.prepCondition() {
-
 		if err := r.prep(); err != nil {
-			return err
+			// ignore flush failures
+			if !errors.Is(err, ErrFlushShardsFailed) {
+				log.Error(err, "failed to flush shards",
```

Comment:
this sort of goes against the pattern idea of this generic function... could we move this up into the functions that generate it?